packages/core/package-manager/src/NodeResolverBase.js

```diff
@@ -126,12 +162,42 @@ export class NodeResolverBase<T> {
     if (moduleDir) {
       return {
         moduleName,
-        subPath: path.join(...parts),
+        subPath,
         moduleDir: moduleDir,
-        filePath: parts.length > 0 ? path.join(moduleDir, ...parts) : moduleDir,
+        filePath: subPath ? path.join(moduleDir, subPath) : moduleDir,
       };
     }
 
     return null;
   }
+
+  getNodeModulesPackagePath(sourceFile: FilePath): ?FilePath {
+    // If the file is in node_modules, we can find the package.json in the root of the package
+    // by slicing from the start of the string until 1-2 path segments after node_modules.
```

Comment:
Not necessarily, e.g. https://unpkg.com/browse/preact@10.5.12/debug/

Reply:
Yeah... in this case I think we want the top-level package to get the version?

Reply:
I guess it doesn't make a difference here because the node resolver doesn't use `package.json#main` anyway